api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -578,3 +583,22 @@ message LayeredRuntime {
   // such that later layers in the list overlay earlier entries.
   repeated RuntimeLayer layers = 1;
 }
+
+// Used to specify the header that needs to be registered as an inline header.
+// Inline headers provide O(1) search performance, but each inline header imposes
+// an additional memory overhead on all instances of the corresponding type of
+// HeaderMap or TrailerMap.
+message CustomInlineHeader {
+  enum InlineHeaderType {
+    REQUEST_HEADER = 0;
+    REQUEST_TRAILER = 1;
+    RESPONSE_HEADER = 2;
+    RESPONSE_TRAILER = 3;
+  }
+
+  // The name of the header that is expected to be set as the inline header.
+  string inline_header_name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // The type of the header that is expected to be set as the inline header.
+  InlineHeaderType inline_header_type = 2;
```

Comment:
I think that the following PGV constraint `(validate.rules).enum.defined_only = true` can be added.

Reply:
👌